data/topic-filter.go

```diff
@@ -0,0 +1,213 @@
+	LabourMarketAndTravelToWork = Topic{
+		LocaliseKeyName: "LabourMarketAndTravelToWork",
+		Group:           "labour_market_and_travel_to_work",
+		SubTopics:       []string{"labour_market_and_travel_to_work"},
+		ShowInWebUI:     true,
+	}
+
+	// SexualOrientationAndGenderIdentity - Search information specific for sexual orientation and gender identity
+	SexualOrientationAndGenderIdentity = Topic{
+		LocaliseKeyName: "SexualOrientationAndGenderIdentity",
+		Group:           "sexual_orientation_and_gender_identity",
+		SubTopics:       []string{"sexual_orientation_and_gender_identity"},
+		ShowInWebUI:     true,
+	}
+
+	// Veterans - Search information specific for Veterans
+	Veterans = Topic{
+		LocaliseKeyName: "Veteran",
+		Group:           "veterans",
+		SubTopics:       []string{"veterans"},
+		ShowInWebUI:     true,
+	}
+
+	// topicFilterOptions contains all the possible filter available on the search page
+	topicFilterOptions = map[string]Topic{
+		DemographyAndMigration.Group:                 DemographyAndMigration,
+		Education.Group:                              Education,
+		EthnicGroupNationalIdentityAndReligion.Group: EthnicGroupNationalIdentityAndReligion,
+		HealthDisabilityAndUnpaidCare.Group:          HealthDisabilityAndUnpaidCare,
+		Housing.Group:                                Housing,
+		LabourMarketAndTravelToWork.Group:            LabourMarketAndTravelToWork,
+		SexualOrientationAndGenderIdentity.Group:     SexualOrientationAndGenderIdentity,
+		Veterans.Group:                               Veterans,
+	}
+)
+
+// reviewFilter retrieves filters from query, checks if they are one of the filter options, and updates validatedQueryParams
+func reviewTopicFilters(ctx context.Context, urlQuery url.Values, validatedQueryParams *SearchURLParams) error {
+	topicFilters := urlQuery["topics"]
+
+	if topicFilters == nil {
+		return nil
+	}
+
+	topics := strings.Split(topicFilters[0], ",")
+
+	for _, topicFilterQuery := range topics {
+
+		topicFilterQuery = strings.ToLower(topicFilterQuery)
+
+		if topicFilterQuery == "" {
+			continue
+		}
+
+		topicFilter, found := topicFilterOptions[topicFilterQuery]
+
+		if !found {
+			err := errs.ErrFilterNotFound
+			logData := log.Data{"topic filter not found": topicFilter}
+			log.Error(ctx, "failed to find topic filter", err, logData)
+
+			return err
+		}
+
+		validatedQueryParams.TopicFilter.Query = append(validatedQueryParams.TopicFilter.Query, topicFilter.Group)
+		validatedQueryParams.TopicFilter.LocaliseKeyName = append(validatedQueryParams.TopicFilter.LocaliseKeyName, topicFilter.LocaliseKeyName)
+	}
+
+	return nil
+}
+
+// GetTopicCategories returns all the categories and its content types where all the count is set to zero
+func GetTopicCategories() []TopicCategory {
+	var topicCategories []TopicCategory
+	topicCategories = append(topicCategories, TopicCategories...)
+
+	// To get a different reference of Topic - deep copy
+	for i, topicCategory := range topicCategories {
```

Comment:
Suggest change this to:

```suggestion
	for i := range topicCategories {
```

And then anywhere that uses `topicCategory` change to , `topicCategories[i]`